From a8c6e26b2cf413f27a177b5ec3e1deaaf27748e1 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Mon, 4 Sep 2017 10:31:55 +0800
Subject: [PATCH 2495/5242] MLK-16353-2 video: fbdev: dcss: non-fourcc format
 should not pass var check

commit  3adc46304d75f637849e349817aeca8da1407397 from
https://source.codeaurora.org/external/imx/linux-imx.git

The non-fourcc pixel format for framebuffer is not supported
yet, so the var check cannot pass in this case. Add false
return in this case.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index caca20c..d6b21ec 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -2419,7 +2419,6 @@ static int dcss_check_var(struct fb_var_screeninfo *var,
 
 	switch (var->grayscale) {
 	case 0:		/* TODO: color */
-		break;
 	case 1:		/* grayscale */
 		return -EINVAL;
 	default:	/* fourcc */
-- 
1.7.9.5

